### PR TITLE
Fixes a fuck ton more harddels

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request
 Redoes how geese handle eating shit, it was fucking stupid and caused harddels, and while this method is technically slower in the best case, it's a fucking goose
Fixes action related harddels, I hate how they work but at least this way they won't hold refs.
Fixes the hierophont causing its beacon to harddel
Removes the M variable from megafauna actions, it was used like a typed owner and caused harddels, so I burned it
Fixes target and targets_from harddels, replaces all setters of target with LoseTarget and GiveTarget, which should help maintain behavior. I'm not sure if this breaks anything, but if it does we should fix the assumptions that code makes instead of reverting this change
Fixes more area_senstive_contents related harddels, we need to allow the mob to move before clearing out its list.
Fixes marked object harddels (I'm coming for you admin team)
Fixes a language based human harddel
Fixes managed overlay related harddels (This was just emissive blockers, but I think this is a good safety net to have. If we clear the overlay list we should clear this one as well)
Fixes bot core harddels, I hate the fact that this exists but it has no reason to know who its owner is
 Adds a walk(src, 0) to simple_animal destroy, it's the best bang for the buck in terms of stopping spurious harddels. Walk related harddels aren't that expensive in the first place, since byond does the same thing I'm doing here, but this makes finding mob harddels easier, so let's go with it
I fixed another source of part harddels, I hate fullupgrade so much
Fixes all the sound loop harddels 

## Why It's Good For The Game

I want more processing time god damn it

## Changelog
:cl:
code: Td should spike slightly less, assuming harddels actually go down
code: You feel a great disturbance in the force, mob code has changed. Please report any odd targeting behavior on the issue tracker
/:cl:

### CODE REVIEW

**Comment by interestingusernam3** [comment]
‏

File: code/datums/action.dm [hunk 1]
```diff
@@ -72,13 +73,18 @@
 	else
 		Remove(owner)
 
+/datum/action/proc/owner_deleted(datum/source)
```
Comment (Mothblocks) [inline code/datums/action.dm:76]: SIGNAL_HANDLER

File: code/modules/mob/living/simple_animal/bot/bot.dm [hunk 2]
```diff
@@ -897,11 +897,10 @@ Pass a positive integer as an argument to override a bot's default speed.
 /obj/machinery/bot_core
 	use_power = NO_POWER_USE
 	anchored = FALSE
-	var/mob/living/simple_animal/bot/owner = null
 
 /obj/machinery/bot_core/Initialize()
 	. = ..()
-	owner = loc
+	var/mob/living/simple_animal/bot/owner = loc
 	if(!istype(owner))
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/bot/bot.dm:904]:
```suggestion
	if(!isbot(loc))
```

File: code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm [hunk 3]
```diff
@@ -179,16 +179,15 @@
...
-	var/mob/living/simple_animal/hostile/megafauna/M
 	var/chosen_message
 	var/chosen_attack_num = 0
 
 /datum/action/innate/megafauna_attack/Grant(mob/living/L)
-	if(istype(L, /mob/living/simple_animal/hostile/megafauna))
-		M = L
-		return ..()
-	return FALSE
+	if(!istype(L, /mob/living/simple_animal/hostile/megafauna))
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm:186]:
```suggestion
	if(!ismegafauna(L))
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/datums/action.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/datums/action.dm:76]: gottchu

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**